src/modules/AddOnOthers.rb

```diff
@@ -0,0 +1,78 @@
+# encoding: utf-8
+require "yast"
+
+# Yast namespace
+module Yast
+  # This module provides integration of the add-on products
+
+  class AddOnOthersClass < Module
+           
+    include Yast::Logger
+
+    #     add_on_others = [
+    #        {
+    #          "media" => 4, # ID of the source
+    #          "name" : "openSUSE version XX.Y",
+    #          "media_url"=>"dvd:/?devices=/dev/sr1"
+    #          ....
+    #        },
+    #        ...
+    #      ]
+    attr_reader :add_on_others
+
+    def main
+      Yast.import "Pkg"
+      textdomain "packager"      
+
+      @add_on_others = []
+    end
+
+    def Read
+      # Removing all repos which have installed based products
+      # and add-on products.
+      all_products = Pkg.ResolvableProperties("", :product, "")
+      installed_products = all_products.select do |p|
```

Comment:
I'm not sure what you want to achieve here, the `installed_products` actually also contains `:available` products... bad variable name?

Why we simply do not use all repositories which do not contain any product? IIRC the old `AddOnProduct.Export` should export the product repositories, here we need to handle the rest.

Um, and maybe it could beasier if we directly adapted the `AddOnProduct.Export` to export everything...

Reply:
This is used for extracting installed based products too. These products will be not shown in the add-products list.

Reply:
What about something like this:

```ruby
installed_product_names = Y2Packager::Resolvable.find(kind: :product, status: :installed).map(&:name)
installed_available_products = Y2Packager::Resolvable.find(kind: :product, status: :available).select do |p|
  installed_product_names.include?(p.name)
end
```

Note: I'd prefer to rename the variable `installed_products` to `installed_available_products` which better describes the meaning. Please rename it also in the code below. :arrow_down: